satpy/tests/test_demo.py

```diff
@@ -196,6 +197,30 @@ def test_ahi_partial_download(self):
         assert len(files) == 6
 
 
+def test_fci_download(tmp_path, monkeypatch):
+    """Test download of FCI test data."""
+    from satpy.demo import download_fci_test_data
+    monkeypatch.chdir(tmp_path)
+
+    def fake_urlretrieve(url):
+        # create a dummy tarfile
+        fn = tmp_path / "tofu.tar.gz"
+        fn.parent.mkdir(exist_ok=True, parents=True)
+
+        with tarfile.open(fn, mode="x:gz") as tf:
+            for i in range(3):
+                with open(f"fci-rc{i:d}", "w"):
+                    pass
+                tf.addfile(tf.gettarinfo(name=f"fci-rc{i:d}"))
```

Comment:
Can this be moved to a separate function?

Reply:
The part in the with-block or the part in the for-block?  I've never tried to split a nested function, but I assume I could factor it out if needed.

Reply:
Everything that creates the tar file (so the with + for + with blocks, everything I've highlighted here). It only has `fn` as input, right? You move that to a function outside this test or as a staticmethod on the test.

Reply:
Done in 108d69f9a4e2a226699ce79bb5c2d26e1832ba8d